subsys/bluetooth/services/fast_pair/fp_storage/fp_storage_ak.c

```diff
@@ -313,9 +327,62 @@ void fp_storage_ram_clear(void)
 
 	settings_set_err = 0;
 	atomic_set(&settings_loaded, false);
+	reset_prepare = false;
 }
 
 bool bt_fast_pair_has_account_key(void)
 {
-	return (fp_storage_account_key_count() > 0);
+	return (fp_storage_ak_count() > 0);
+}
+
+static void fp_storage_ak_empty_init(void)
+{
+	account_key_next_id = ACCOUNT_KEY_MIN_ID;
+	atomic_set(&settings_loaded, true);
+}
+
+static int ak_delete(uint8_t index)
+{
+	int err;
+	char name[SETTINGS_AK_NAME_MAX_SIZE];
+
+	err = ak_name_gen(name, index);
+	if (err) {
+		return err;
+	}
+
+	err = settings_delete(name);
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+static int fp_storage_ak_reset(void)
+{
+	for (uint8_t index = 0; index < ACCOUNT_KEY_CNT; index++) {
+		int err;
+
+		err = ak_delete(index);
+		if (err) {
+			return err;
+		}
+	}
+
+	fp_storage_ak_ram_clear();
+
+	fp_storage_ak_empty_init();
+
+	return 0;
+}
+
+static void reset_prepare_set(void)
+{
+	reset_prepare = true;
```

Comment:
Should we also set settings loaded to false? If we prepare for reset, the storage data should not be used.

Reply:
@kapi-no Do you agree?

Reply:
yes, that's okay